malaria-app-android/src/main/java/com/peacecorps/malaria/ui/trip_reminder/PlanTripFragment.java

```diff
@@ -0,0 +1,304 @@
+                }
+                return false;
+            }
+
+
+        });
+
+        tvArrivalDate.setOnTouchListener(new OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                final int DRAWABLE_RIGHT = 2;
+
+                if (event.getAction() == MotionEvent.ACTION_UP &&
+                        (event.getRawX() >= (tvArrivalDate.getRight() - tvArrivalDate.getCompoundDrawables()[DRAWABLE_RIGHT].getBounds().width()))) {
+                    checkDepartureOrArrivalEt = false;
+                    // creating show date picker dialog
+                    showDatePicker();
+                    return true;
+                }
+                return false;
+            }
+
+
+        });
+
+        tvReminderTime.setOnTouchListener(new OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                final int DRAWABLE_RIGHT = 2;
+
+                if (event.getAction() == MotionEvent.ACTION_UP &&
+                        (event.getRawX() >= (tvReminderTime.getRight() - tvReminderTime.getCompoundDrawables()[DRAWABLE_RIGHT].getBounds().width()))) {
+                    checkDepartureOrArrivalEt = false;
+                    // creating show date picker dialog
+                    showTimePicker();
+                    return true;
+                }
+                return false;
+            }
+
+
+        });
+    }
+
+    private void showDatePicker() {
+        DatePickerFragment date = new DatePickerFragment();
+
+        if (getFragmentManager() != null) {
+            date.show(getFragmentManager(), "Date Picker");
+        } else {
+            ToastLogSnackBarUtil.showErrorLog("getFragmentManager is null in PlanTripFragment");
+        }
+
+
+        DatePickerDialog.OnDateSetListener ondate = new DatePickerDialog.OnDateSetListener() {
+
+            public void onDateSet(DatePicker view, int year, int month,
+                                  int day) {
+                if (checkDepartureOrArrivalEt) {
+                    dptr = true;
+                    tvDepartureDate.setText(getResources().getString(R.string.et_trip_date, day, month + 1, year));
+                } else {
+                    arr = true;
+                    tvArrivalDate.setText(getResources().getString(R.string.et_trip_date, day, month + 1, year));
+                }
+
+            }
+        };
+        /*
+         * Set Call back to capture selected date
+         */
+        date.setCallBack(ondate);
+
+    }
+
+    private void showTimePicker() {
+        TimePickerFragment timePickerFragment = new TimePickerFragment();
+        if (getFragmentManager() != null && getActivity() != null) {
+            timePickerFragment.show(getActivity().getFragmentManager(), "Time Picker in PlanTripFragment");
+        } else {
+            ToastLogSnackBarUtil.showErrorLog("getFragmentManager is null in PlanTripFragment");
```

Comment:
control will come even if getFragmentManager is not null and getActivity is null. Do you want to take same action (logging error both case). If yes, then correct the logging msg :"getFragmentManager/getActivity is null"